format/bplist/bplist.go

```diff
@@ -0,0 +1,245 @@
+		m := d.FieldU4("type", elementTypeMap)
+		switch m {
+		case elementTypeNullOrBoolOrFill:
+			t := d.U4()
+			switch t {
+			case null:
+				d.FieldValueNil("value")
+			case boolTrue:
+				d.FieldValueBool("value", true)
+			case boolFalse:
+				d.FieldValueBool("value", false)
+			}
+		case elementTypeInt:
+			n := decodeSize(d)
+			d.FieldSFn("value", func(d *decode.D) int64 {
+				return d.UBigIntBE(int(n)).Int64()
+			})
+		case elementTypeReal:
+			n := decodeSize(d)
+			d.FieldFFn("value", func(d *decode.D) float64 {
+				return d.FE(int(n), decode.BigEndian)
+			})
+		case elementTypeDate:
+			d.FieldStrFn("value", func(d *decode.D) string {
+				v := d.F64()
+				t := time.Unix(int64(v), 0)
+				return t.String()
+			})
+		case elementTypeData:
+			d.FieldStruct("value", func(d *decode.D) {
+				n := decodeSize(d)
+				d.FieldRawLen("value", int64(n))
+			})
+		case elementTypeASCIIString:
+			d.FieldStruct("value", func(d *decode.D) {
+				d.FieldStrFn("value", func(d *decode.D) string {
+					n := decodeSize(d)
+					return d.UTF8(int(n))
+				})
+			})
+		case elementTypeUnicodeString:
+			d.FieldStrFn("value", func(d *decode.D) string {
+				n := decodeSize(d)
+				return d.UTF16(int(n))
+			})
+		case elementTypeUID:
+			n := decodeSize(d)
+			d.FieldUFn("value", func(d *decode.D) uint64 {
+				return d.UBigIntBE(int(n)).Uint64()
+			})
+		case elementTypeArray:
+			n := decodeSize(d)
+			i := uint64(0)
+			d.FieldStructArrayLoop("elements", "item",
+				func() bool { return i < n },
+				func(d *decode.D) {
+					idx := d.FieldU8("object_index")
+					decodeReference(d, p, idx)
+					i++
+				})
+		case elementTypeSet:
+			n := decodeSize(d)
+			i := uint64(0)
+			d.FieldArrayLoop("entries",
+				func() bool { return i < n },
+				func(d *decode.D) {
+					idx := d.FieldU8("object_index")
+					decodeReference(d, p, idx)
+					i++
+				})
+		case elementTypeDict:
+			d.FieldStruct("dictionary", func(d *decode.D) {
+				s := decodeSize(d)
+				i := uint64(0)
+				d.FieldStructArrayLoop("entries", "entry",
+					func() bool { return i < s },
+					func(d *decode.D) {
+						var ki, vi uint64
+						ki = d.FieldU8("key_index")
+						d.SeekRel(int64((s-1)*uint64(p.T.objRefSize)*8), func(d *decode.D) {
+							vi = d.FieldU8("value_index")
+						})
+						i++
+						d.FieldStruct("key", func(d *decode.D) {
+							decodeReference(d, p, ki)
+						})
+						d.FieldStruct("value", func(d *decode.D) {
+							decodeReference(d, p, vi)
+						})
+					})
+			})
+		default:
+			d.Errorf("unknown type marker: %d", m)
+		}
+	})
+
+	return nil
+}
+
+func decodeReference(d *decode.D, p *plist, idx uint64) {
+	d.SeekAbs(int64(p.O[idx]*8), func(d *decode.D) {
+		decodeItem(d, p)
+	})
+}
+
+type trailer struct {
+	offTblOffSize    int64
+	objRefSize       int64
+	nObjects         int64
+	topObjectOffset  int64
+	offsetTableStart int64
+}
+
+type plist struct {
+	T *trailer
+	O []uint64
+}
+
+func bplistDecode(d *decode.D, _ any) any {
```

Comment:
Is big endian default endian of this format or is it a bit mixed? default is big endian and default can be changed with `d.Endian = ...` and it will be inheried down

Reply:
It's BE everywhere, I'll make it implicit everywhere.